kernel/aster-nix/src/time/clock.rs

```diff
@@ -0,0 +1,298 @@
+            }))
+        }
+    };
+}
+
+/// Generate code for the declaration and initialization of the supported clock and global timer manager.
+///
+/// This macro accept a series of triples: `ClockID`, `Clock Name`, `Has Global TimerManager`(true or false).
+/// It will declare a list that contains all supported `ClockID`, a instance for each clock,
+/// and a `TimerManager` for corresponding clock whose `Has Global TimerManager` is set true.
+///
+/// Then this macro will generate the corresponding initialization function for clocks and timer managers,
+/// and two functions to fetch the clock instances and timer managers from `ClockID` respectively.
+macro_rules! init_supported_global_clock_and_timer_manager {
+    ($($clock_id:ident, $clock_type:ty, $has_manager:ident;)*) => {
+        /// A list of all supported clock IDs for time-related functions.
+        const ALL_SUPPORTED_GLOBAL_CLOCK_IDS: [ClockID; 6] = [
+            $(ClockID:: $clock_id,)*
+        ];
+
+        $(
+            paste! {
+                pub static [<$clock_id _INSTANCE>]: Once<Arc<dyn Clock>> = Once::new();
+            }
+            init_global_timer_manager!($clock_id, $has_manager, declare);
+        )*
+
+        fn _init_global_clock_and_timer_manager() {
+            $(
+                let clock = Arc::new(<$clock_type>::default());
+                paste! {
+                    [<$clock_id _INSTANCE>].call_once(|| clock.clone());
+                }
+                init_global_timer_manager!($clock_id, $has_manager, init, clock);
+            )*
+        }
+
+        fn _id_to_global_manager(clock_id: &ClockID) -> Option<&Arc<TimerManager>> {
+            match clock_id {
+                $(
+                    ClockID::$clock_id => init_global_timer_manager!($clock_id, $has_manager, function),
+                )*
+                _ => None,
+            }
+        }
+
+        fn _id_to_global_clock(clock_id: &ClockID) -> Option<&Arc<dyn Clock>> {
+            match clock_id {
+                $(
+                    ClockID::$clock_id => paste!{Some(&[<$clock_id _INSTANCE>].get().unwrap())},
+                )*
+                _ => None,
+            }
+        }
+    };
+}
+
+init_supported_global_clock_and_timer_manager! {
+    //  ClockID              |    Clock Names       |   Has Global TimerManager
+    CLOCK_REALTIME,            RealTimeClock,           true;
+    CLOCK_REALTIME_COARSE,     RealTimeCoarseClock,     false;
+    CLOCK_MONOTONIC,           MonotonicClock,          true;
+    CLOCK_MONOTONIC_COARSE,    MonotonicCoarseClock,    false;
+    CLOCK_MONOTONIC_RAW,       MonotonicRawClock,       false;
+    CLOCK_BOOTTIME,            BootTimeClock,           false;
+}
+
+/// Get a list that contains all supported global`ClockID`s.
+pub fn all_supported_global_clock_ids() -> &'static [ClockID] {
+    &ALL_SUPPORTED_GLOBAL_CLOCK_IDS
+}
+
+/// Get the instance of the global clock depends on the input `ClockID`.
```

Comment:
I think "global clocks" is quite a vague term. Does "global" mean "world-wide"? Or does "global" mean a public static variable that is globally accessible by all the kernel code? And what is the opposite of "global"? A "local" clock? What would a "local" clock look like? So you see: mentioning a term ("global clock") without defining it first makes people wonder. Not a good thing for clarity.

So I suggest replacing "global clocks" with "system-wide clocks". The latter should be sufficiently self-describing.

In fact, I don't think the `time` module needs to mention this term "global clocks" at all. The only seemingly legit use case of `id_to_global_clock` is `now_as_duration`, which is used to implement `sys_clock_gettime` and `clock_nanosleep`. So the mapping from clock IDs to clock types/instances can be done in `clock_gettime.rs`, instead of the `time` module. This way, you don't need to bother about the so-called global clocks in the `time` module.

Reply:
`system-wide clocks` is a better name. After replacing `global clocks` with `system-wide clocks`, I think `time` module can remain the function `get_system_wide_clock(clock_id)` since it will not result in ambiguity and ClockID is just defined in `time` module. This mapping function will act as one of the Linux-oriented policies.

This function will be used in `sys_clock_gettime`, `sys_clock_nanosleep` and future in `sys_timer_create`. Putting this function in `clock_gettime.rs` will let these implementations use a function from a syscall-implementation module, which is kind of weird.